sycl/include/sycl/detail/property_helper.hpp

```diff
@@ -46,8 +46,9 @@ enum DataLessPropKind {
   QueueSubmissionImmediate = 21,
   GraphAssumeDataOutlivesBuffer = 22,
   GraphAssumeBufferOutlivesGraph = 23,
+  GraphNodeLeavesDependencies = 24,
```

Comment:
```suggestion
  GraphDependOnAllLeaves = 24,
```
Nit but I think this was a little ambiguous and I think this is more aligned with the actual prop name and what it does.